extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/BaseFileFormatTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.hadoop.file;
+
+import com.hazelcast.function.ConsumerEx;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.file.FileSourceBuilder;
+import com.hazelcast.jet.pipeline.test.Assertions;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@RunWith(Parameterized.class)
+public abstract class BaseFileFormatTest extends JetTestSupport {
+
+    @Parameter
+    public boolean useHadoop;
+
+    @Parameters(name = "{index}: useHadoop={0}")
+    public static Iterable<?> parameters() {
+        return Arrays.asList(true, false);
+    }
+
+    @SafeVarargs
+    protected final <T> void assertItemsInSource(FileSourceBuilder<T> source, T... items) {
+        assertItemsInSource(source, collected -> assertThat(collected).containsOnly(items));
+    }
+
+    protected <T> void assertItemsInSource(
+            FileSourceBuilder<T> source, ConsumerEx<List<T>> assertion
+    ) {
+        assertItemsInSource(1, source, assertion);
+    }
+
+    protected <T> void assertItemsInSource(
+            int memberCount, FileSourceBuilder<T> source, ConsumerEx<List<T>> assertion
+    ) {
+        if (useHadoop) {
+            source.useHadoopForLocalFiles(true);
+        }
+
+        Pipeline p = Pipeline.create();
+
+        p.readFrom(source.build())
+         .apply(Assertions.assertCollected(assertion));
+
+        JetInstance[] jets = createJetMembers(memberCount);
+        jets[0].newJob(p).join();
+
+        for (JetInstance jet : jets) {
```

Comment:
this is not necessary, the instances are terminated at the end of the test already. 
we should also move the instance creation to a setup method so that new instances are not created for each call of this method.

Reply:
The `com.hazelcast.jet.pipeline.test.Assertions#assertCollected` checks against all collected items, so you actually need a new instance everytime. Then if the instances are not shut down they connect with instances from previous calls.